src/client/views/cube-view/api-context.tsx

```diff
@@ -17,58 +17,91 @@
 import { Dataset, DatasetJS } from "plywood";
 import React, { useContext } from "react";
 import { ClientAppSettings } from "../../../common/models/app-settings/app-settings";
+import { Dimension } from "../../../common/models/dimension/dimension";
 import { Essence } from "../../../common/models/essence/essence";
-import { Unary } from "../../../common/utils/functional/functional";
+import { Binary, Unary } from "../../../common/utils/functional/functional";
 import { DEFAULT_VIEW_DEFINITION_VERSION, definitionConverters } from "../../../common/view-definitions";
 import { Ajax } from "../../utils/ajax/ajax";
 
-export type QueryCall = Unary<Essence, Promise<Dataset>>;
+export type VisualizationQuery = Unary<Essence, Promise<Dataset>>;
 
-interface ApiContextValue {
-  query: QueryCall;
+export type BooleanFilterQuery = Binary<Essence, Dimension, Promise<Dataset>>;
+
+export interface ApiContextValue {
+  visualizationQuery: VisualizationQuery;
+  booleanFilterQuery: BooleanFilterQuery;
 }
 
-const InternalApiContext = React.createContext<ApiContextValue>({
-  get query(): QueryCall {
+export const ApiContext = React.createContext<ApiContextValue>({
+  get booleanFilterQuery(): BooleanFilterQuery {
+    throw new Error("Attempted to consume ApiContext when there was no Provider");
```

Comment:
Code is duplicated. I would see custom exception for all cases when provider doesn't exist.